### PR TITLE
Story/sc 1340 update poetry

### PR DESCRIPTION
**Pull request by Fabian6781**
## Jira Story
Story/SC 1340

## About
Update poetry to version 1.5.1

## Associated Pull Requests
N/A

## Update Changelog

*   [X] Changelog updated

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (gordon-epc) [inline .github/workflows/ci.yml:1]: `actions/setup-python@v2` -> `actions/setup-python@v4`

File: pyproject.toml [hunk 1]
```diff
@@ -7,10 +7,10 @@ license = "MIT"
 readme = "README.rst"
 
 [tool.poetry-dynamic-versioning]
-#set to false, poetry-dynamic-versioning used only in CI
-enable = false
+#set to false.  poetry-dynamic-versioning only needed for CI versioning.
```
Comment (gordon-epc) [inline pyproject.toml]: The comment doesn't make sense anymore.
```suggestion
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: pyproject.toml
Comment (gordon-epc) [inline pyproject.toml:1]:
```
# For development and git commit, the __version__ variable below is set
# The version variable below is set to the generic build placeholder "0.0.0".
# (In other words: leave alone!) During CI, poetry dynamic versioning
# modifies the __version__ variable in the applicable __init__.py file.
version = "0.0.0"
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: pyproject.toml
Comment (Fabian6781) [reply pyproject.toml:1]: Done.

File: pyproject.toml (hunk 1 above)
Comment (Fabian6781) [reply pyproject.toml]: Resolved.

File: .github/workflows/ci.yml
Comment (Fabian6781) [reply .github/workflows/ci.yml:1]: Done.

File: pyproject.toml [hunk 2]
```diff
@@ -1,16 +1,19 @@
 [tool.poetry]
 name = "sunspecdemo"
+# For development and git commit, the __version__ variable below is set
```
Comment (gordon-epc) [inline pyproject.toml]: Not sure where the extra line came from?
```suggestion
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -41,4 +42,4 @@ sunspecdemo = 'sunspecdemo.cli:cli'
 
 [build-system]
 requires = ["setuptools", "poetry-core>=1.0.0", "poetry-dynamic-versioning",]
-build-backend = "poetry.core.masonry.api"
```
Comment (epcAnker) [inline pyproject.toml:44]: did the masonry.api do anything that we need?

**Review by epcAnker — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (gordon-epc) [reply pyproject.toml:44]: The `poetry_dynamic_versioning.backend` is a wrapper (they call it a "patch") for `poetry.core.masonry.api` to allow poetry dynamic versioning to function.

File: pyproject.toml [hunk 4]
```diff
@@ -30,15 +32,14 @@ toml = "^0.10.2"
...
 pyinstaller = "4.4"
 
 
 [tool.poetry.scripts]
 sunspecdemo = 'sunspecdemo.cli:cli'
 
 [build-system]
-requires = ["setuptools", "poetry-core>=1.0.0", "poetry-dynamic-versioning",]
-build-backend = "poetry.core.masonry.api"
+requires = ["setuptools", "poetry-core>=1.0.0", "poetry-dynamic-versioning>=1.0.0,<2.0.0",]
```
Comment (epcAnker) [inline pyproject.toml]: format

Comment (Fabian6781) [reply pyproject.toml]: Done.

**Pull request closed without merging**